src/main/java/com/amazon/opendistroforelasticsearch/jdbc/PreparedStatementImpl.java

```diff
@@ -75,11 +75,19 @@ public PreparedStatementImpl(ConnectionImpl connection, String sql, Logger log)
     public ResultSet executeQuery() throws SQLException {
         log.debug(() -> logEntry("executeQuery()"));
         checkOpen();
-        ResultSet rs = executeQueryX();
+        ResultSet rs = executeQueryXWithFetchSize(getFetchSize());
         log.debug(() -> logExit("executeQuery", rs));
         return rs;
     }
 
+    protected ResultSet executeQueryXWithFetchSize(int fetchSize) throws SQLException {
```

Comment:
Question, why adding executeQueryXWithFetchSize instead of changing executeQueryX?

Reply:
I thought of not breaking existing callers using `executeQueryX`. But yes it should be ok changed `executeQueryX ` instead of adding `executeQueryXWithFetchSize `.

Done.